### DISCUSSION TITLE
Scraping dynamic targets

### DISCUSSION THREAD

**Discussion by SebiBurchi** [prometheus-operator/prometheus-operator#4084] [Q&A]
Hello,
I am trying to use Prometheus-operator in order to collect some metrics.
I saw that prom-operator only uses secret file for additional scrape configs.
Here i have a problem because i need to modify that secret inside the pod from my Kubernetes cluster, and the changes to be propagated to prom-operator scrape targets. For this thing to happen, I tried to mount the secret file as a volume in my pod file system. But, after last changes in Kubernetes project, configMap and secrets volumeMount are always mounted readOnly.
So, for the moment i am stuck because prom-operator only uses secret file, and the secret file can not longer be modified after it was mounted.
So, how can i use prom-operator for scraping dynamic targets in this case?

Link to Kubernetes changes: https://github.com/kubernetes/kubernetes/issues/62099

**Comment by paulfantom** [comment]
You can use `ServiceMonitor` or `PodMonitor` CRs for this and let prometheus-operator modify secrets and reload prometheus :)

**Comment by SebiBurchi** [comment]
> You can use `ServiceMonitor` or `PodMonitor` CRs for this and let prometheus-operator modify secrets and reload prometheus :)

Actually, the dynamic targets are external devices. So, basically, i want to to add device ip as a dynamic target. The device is not running inside the cluster. We are talking about external targets which are agnostic of k8s cluster. So, i don't think i can use ServiceMonitor or PodMonitor

**Comment by paulfantom** [comment]
You still can do this by creating `Endpoint` objects, sth like in https://devops.college/prometheus-operator-how-to-monitor-an-external-service-3cb6ac8d5acb.

A different approach would be to use `additionalScrapeConfigs` field of [PrometheusSpec](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec)

**Comment by SebiBurchi** [comment]
> You still can do this by creating `Endpoint` objects, sth like in https://devops.college/prometheus-operator-how-to-monitor-an-external-service-3cb6ac8d5acb.
> 
> A different approach would be to use `additionalScrapeConfigs` field of [PrometheusSpec](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec)

Actually, this is my real problem: `additionalScrapeConfigs` uses secret key for specifying targets. How i said, i need to modify that secret inside a pod and i can no longer do this in kubernetes.
